ts/WoltLabSuite/Core/Ui/ItemList/LineBreakSeparatedText.ts

```diff
@@ -0,0 +1,258 @@
+/**
+ * UI element that shows individual lines of text as distinct list items but saves them as line
+ * break-separated text.
+ *
+ * @author  Matthias Schmidt
+ * @copyright  2001-2021 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/ItemList/LineBreakSeparatedText
+ */
+
+import * as UiConfirmation from "../Confirmation";
+import * as Language from "../../Language";
+import DomUtil from "../../Dom/Util";
+
+export interface LineBreakSeparatedTextOptions {
+  listItemId: string;
+  submitFieldName: string;
+}
+
+export class UiItemListLineBreakSeparatedText {
+  protected clearButton?: HTMLAnchorElement = undefined;
+  protected itemInput?: HTMLInputElement = undefined;
+  protected readonly itemList: HTMLElement;
+  protected items = new Set<string>();
+  protected readonly options: LineBreakSeparatedTextOptions;
+
+  constructor(options: LineBreakSeparatedTextOptions) {
+    this.options = options;
+
+    this.itemList = document.getElementById(this.options.listItemId)!;
+    if (!this.itemList) {
+      throw new Error(`Unknown element with id '${this.options.listItemId}'`);
+    }
```

Comment:
In recent API design I prefer passing the element directly, instead of passing an ID. The caller can more easily identify the element.

see also: https://github.com/WoltLab/WCF/blob/b6128f977cbec78c575c9ee2fd66799764023251/ts/WoltLabSuite/Core/Ui/User/Multifactor/Totp/Qr.ts#L3